### PR TITLE
OOP school library: refactor your code

### PR DESCRIPTION
**Pull request by TesheMaximillan**
> In this milestone, we tried to refactor♻️ our code according to the following principles

## Project requirements 📌
***

* we followed YAGNI, KISS, and DRY rules, and the Ruby style guide is respected.
* we followed SOLID principles.
* we gave special attention to the "Single-responsibility Principle".
* The role of main.rb is only to start the code and ask the user for the initial choice of options (or to quit).
* we used a utility Module to extract some pieces of logic

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -17,17 +17,6 @@ def initialize
...
-    45.times { print '-' }
-    puts "\n"
-    print "1. List all books\n2. List all people\n3. Create a person\n" \
-          "4. Create a book\n5. Create a rental\n6. List all rentals for" \
-          "a given person id\n7. Exit\n"
-  end
-
   def create_book
     print 'Title: '
     title = gets.chomp
```
Comment (thecodechaser) [inline app.rb]: - [ ] In this method you are getting input from users and creating the book, This method does not follow `single responsibility principle`. Please make sure to separate these operations into different methods to achieve `single responsibility principle`.

File: app.rb [hunk 2]
```diff
@@ -75,18 +64,20 @@ def create_person
...
-    person.each { |p| puts "#{type} Name: #{p.name}\t\tID: #{p.id}\t\tAge: #{p.age}" }
+    puts
+    person.each do |p|
+      type = '[Student]'
+      type = '[Teacher]' if p.instance_of?(Teacher)
+      puts "#{type} Name: #{p.name}\t\tID: #{p.id}\t\tAge: #{p.age}"
+    end
   end
 
   def create_rental
```
Comment (thecodechaser) [inline app.rb]: - [ ] In this method you are getting input from users and creating the rental, This method does not follow `single responsibility principle`. Please make sure to separate these operations into different methods to achieve `single responsibility principle`.

File: app.rb [hunk 3]
```diff
@@ -75,18 +64,20 @@ def create_person
     create_person_object(select_person)
```
Comment (thecodechaser) [inline app.rb:76]: - [ ] In `create_person_object` method you are getting input from users and creating the person, This method does not follow `single responsibility principle`. Please make sure to separate these operations into different methods to achieve `single responsibility principle`.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @TesheMaximillan, and Team👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @TesheMaximillan 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Well done making the required changes from the previous review.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**